LiteFarmOrg/LiteFarm
# Lf 4357 concurrency issue the finance page will be broken once the user will try to change the expense revenue type of the already deleted revenue or expense

**Description**

The finance page is broken if the user changes the expense/revenue type to already deleted revenue or expense. This behaviour should be prevented and an error message should be displayed in this case.

Jira link:https://lite-farm.atlassian.net/browse/LF-4357

**Type of change**


**How Has This Been Tested?**

1. Create a new “Expense" type
2. Open the newly created “Expense" type in different tab
3. Edit any previously created Expense.
4. Delete the Expense type
5. Change the type of the Expense to the one just created
6. An error message is shown while preventing user to change to the deleted Expense type

Follow the same steps for Revenue as well to test the functionality.


**Checklist:**
